code/modules/mob/living/carbon/human/_species.dm

```diff
@@ -256,20 +256,24 @@ GLOBAL_LIST_EMPTY(features_by_species)
  * Arguments:
  * * gender - The gender that the name should adhere to. Use MALE for male names, use anything else for female names.
  * * unique - If true, ensures that this new name is not a duplicate of anyone else's name currently on the station.
- * * lastname - Does this species' naming system adhere to the last name system? Set to false if it doesn't.
+ * * last_name - Do we use a given last name or pick a random new one?
+ * * first_name - Do we use a given first name or pick a random new one?
  */
-/datum/species/proc/random_name(gender,unique,lastname)
+/datum/species/proc/random_name(gender, unique, last_name, first_name)
```

Comment:
Please update the children proc with the new arg you added (In fact this will probably not work for species which implement their own proc like lizards)